packages/client/lib/sync/fetcher/headerfetcher.ts

```diff
@@ -28,13 +33,17 @@ export class HeaderFetcher extends BlockFetcher {
    * Requests block headers for the given task
    * @param job
    */
-  async request(job: any) {
+  async request(job: Job<JobTask, BlockHeaderResult>) {
     const { task, peer } = job
-    if (this.flow.maxRequestCount(peer, 'GetBlockHeaders') < this.maxPerRequest) {
+    if (this.flow.maxRequestCount(peer!, 'GetBlockHeaders') < this.maxPerRequest) {
       // we reached our request limit. try with a different peer.
-      return false
+      return undefined
```

Comment:
is `return` and `return undefined` the same?

Reply:
Yep, fixed.